1.19.0rc2: tests failure on armhf

Hello,
running tests for numpy 1.19.0rc2 fails on arch `armhf` with this failure (both for plain and debug flavors):

```
=================================== FAILURES ===================================
_______________________ Test_ARM_Features.test_features ________________________

self = <numpy.core.tests.test_cpu_features.Test_ARM_Features object at 0xb2a0b0e8>

    def test_features(self):
        self.load_flags()
        for gname, features in self.features_groups.items():
            test_features = [self.cpu_have(f) for f in features]
>           assert_equal(__cpu_features__.get(gname), all(test_features))
E           AssertionError: 
E           Items are not equal:
E            ACTUAL: True
E            DESIRED: False

features   = ['NEON', 'HALF']
gname      = 'NEON_FP16'
self       = <numpy.core.tests.test_cpu_features.Test_ARM_Features object at 0xb2a0b0e8>
test_features = [False, True]

tmp/usr/lib/python3/dist-packages/numpy/core/tests/test_cpu_features.py:20: AssertionError
```

full build log [here](https://buildd.debian.org/status/fetch.php?pkg=numpy&arch=armhf&ver=1%3A1.19.0%7Erc2-2&stamp=1591497276&raw=0). It was not failing with 1.18.4

--- a/numpy/core/src/common/npy_cpu_features.c.src
+++ b/numpy/core/src/common/npy_cpu_features.c.src
@@ -367,9 +367,10 @@ npy__cpu_init_features_linux(void)
         npy__cpu_init_features_arm8();
     } else {
         npy__cpu_have[NPY_CPU_FEATURE_NEON]       = (hwcap & NPY__HWCAP_NEON)   != 0;
-        npy__cpu_have[NPY_CPU_FEATURE_NEON_FP16]  = (hwcap & (NPY__HWCAP_NEON | NPY__HWCAP_VFPv3 |
-                                                              NPY__HWCAP_HALF)) != 0;
-        npy__cpu_have[NPY_CPU_FEATURE_NEON_VFPV4] = (hwcap & (NPY__HWCAP_NEON | NPY__HWCAP_VFPv4)) != 0;
+        if (npy__cpu_have[NPY_CPU_FEATURE_NEON]) {
+            npy__cpu_have[NPY_CPU_FEATURE_NEON_FP16]  = (hwcap & NPY__HWCAP_HALF) != 0;
+            npy__cpu_have[NPY_CPU_FEATURE_NEON_VFPV4] = (hwcap & NPY__HWCAP_VFPv4) != 0;
+        }
     }
     return 1;
 }

